test/new_relic/agent/configuration/default_source_test.rb

```diff
@@ -291,6 +285,81 @@ def test_allowlist_blocks_invalid_values_and_uses_a_default
       end
     end
 
+    def test_boolean_configs_accepts_yes_on_and_true_as_strings
+      key = :'send_data_on_exit'
+      default = ::NewRelic::Agent::Configuration::DefaultSource.default_for(key)
+      config_array = %w[yes on true]
+
+      config_array.each do |value|
+        with_config(key => value) do
+          assert NewRelic::Agent.config[key]
```

Comment:
This (nice, DRY) approach to looping over test inputs stored in an array can potentially lead to confusion over which item(s) in the array are actually causing the problem if there are ever any errors in the future.

Sometimes it's worth going against DRY in the tests for this reason. Sometimes it's worth keeping the testing code clean and just adding some extra context to help the tester. I'm thinking that would be a good approach to take here:

```suggestion
          assert NewRelic::Agent.config[key], "The '#{value}' value failed to evaluate as truthy!"
```

Reply:
Nice- great feedback! Made this change https://github.com/newrelic/newrelic-ruby-agent/pull/2847/commits/be01eafceca9862380560b212346bac17f6db246